Source/WebCore/Modules/mediasource/MediaSourceRegistry.h

```diff
@@ -47,13 +47,15 @@ class MediaSourceRegistry final : public URLRegistry {
     static MediaSourceRegistry& registry();
 
     // Registers a blob URL referring to the specified media source.
-    void registerURL(ScriptExecutionContext&, const URL&, URLRegistrable&)final;
-    void unregisterURL(const URL&) final;
+    void registerURL(const ScriptExecutionContext&, const URL&, URLRegistrable&) final;
+    bool unregisterURL(const URL&) final;
+    void unregisterURLsForContext(const ScriptExecutionContext&) final;
     URLRegistrable* lookup(const String&) const final;
 
 private:
     MediaSourceRegistry();
-    MemoryCompactRobinHoodHashMap<String, RefPtr<MediaSource>> m_mediaSources;
+    MemoryCompactRobinHoodHashMap<String, std::pair<RefPtr<MediaSource>, ScriptExecutionContextIdentifier>> m_mediaSources;
```

Comment:
Can the pairs use `Ref` instead of 15th RefPtr`?

Reply:
I'll try. I'm not quite sure how it works when you call HashMap::get() and want to make sure the value is valid (i.e. the key was in the map). Currently we null check the media source but if it becomes a Ref then that no longer works.

Reply:
Oh, it seems I can use Ref<> and then check Ref<>::isHashTableEmptyValue() after calling HashMap::get() / HashMap::take().

Reply:
I sort of regret suggesting it if that's how we have to use it. But either way is of course OK.